lib/puppet/provider/base_dsc/powershell.rb

```diff
@@ -15,6 +17,49 @@
 Applies DSC Resources by generating a configuration file and applying it.
 EOT
 
+  def initialize(value={})
+    super(value)
+    set_symlink_folder
+  end
+
+  if Puppet::Util::Platform.windows?
```

Comment:
Originally I set this around the method call, but moved it out once I needed access to the registry.